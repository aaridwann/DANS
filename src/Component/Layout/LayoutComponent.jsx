import React from "react";
import NavbarComponent from "../Navbar/NavbarComponent";
import SlideShow from "../SlideShow/SlideShow";
import LoadingComponent from "../LoadingComponent/LoadingComponent";

import ToastComponent from "../Toast/ToastComponent";

function LayoutComponent({ children }) {
  return (
    <div className=" w-full bg-slate-50 h-screen">
      <LoadingComponent />
      <div className="z-50 transition-all fixed px-4 w-full overflow-y-auto h-full">
        <ToastComponent />
        <NavbarComponent />
        {children}
      </div>
      <div className=" w-full md:h-80 h-52 bg-gradient-to-r from-cyan-500 overflow-hidden to-blue-500 fixed top-0 items-end justify-end ">
        <SlideShow />
      </div>
    </div>
  );
}

export default LayoutComponent;
