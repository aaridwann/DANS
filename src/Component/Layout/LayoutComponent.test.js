import React from "react";
import { render } from "@testing-library/react";
import LayoutComponent from "./LayoutComponent";
import "@testing-library/jest-dom/extend-expect";

jest.mock("../LoadingComponent/LoadingComponent", () => {
  return function MockLoadingComponent() {
    return <div>LoadingComponent</div>;
  };
});
jest.mock("../Toast/ToastComponent", () => () => <div>ToastComponent</div>);
jest.mock("../Navbar/NavbarComponent", () => () => <div>NavbarComponent</div>);
jest.mock("../SlideShow/SlideShow", () => () => <div>SlideShow</div>);

describe("LayoutComponent", () => {
  it("renders correctly", () => {
    const children = <div>Children Component</div>;
    const { container } = render(
      <LayoutComponent loading={false}>{children}</LayoutComponent>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  
  it("should render LoadingComponent and children", () => {
    const children = <div>Children Component</div>;
    const { getByText } = render(<LayoutComponent>{children}</LayoutComponent>);
    expect(getByText("LoadingComponent")).toBeInTheDocument();
    expect(getByText("Children Component")).toBeInTheDocument();
  });

  test("should not render LoadingComponent when loading is false", () => {
    const children = <div>Children Component</div>;
    const { queryByText } = render(
      <LayoutComponent loading={false}>{children}</LayoutComponent>
    );
    expect(queryByText("LoadingComponent")).toBeInTheDocument();
  });

  it("should render ToastComponent, NavbarComponent and SlideShow", () => {
    const children = <div>Children Component</div>;
    const { getByText } = render(<LayoutComponent>{children}</LayoutComponent>);
    expect(getByText("ToastComponent")).toBeInTheDocument();
    expect(getByText("NavbarComponent")).toBeInTheDocument();
    expect(getByText("SlideShow")).toBeInTheDocument();
  });
});
