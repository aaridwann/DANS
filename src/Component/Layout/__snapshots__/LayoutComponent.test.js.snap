// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent renders correctly 1`] = `
<div
  class=" w-full bg-slate-50 h-screen"
>
  <div>
    LoadingComponent
  </div>
  <div
    class="z-50 transition-all fixed px-4 w-full overflow-y-auto h-full"
  >
    <div>
      ToastComponent
    </div>
    <div>
      NavbarComponent
    </div>
    <div>
      Children Component
    </div>
  </div>
  <div
    class=" w-full md:h-80 h-52 bg-gradient-to-r from-cyan-500 overflow-hidden to-blue-500 fixed top-0 items-end justify-end "
  >
    <div>
      SlideShow
    </div>
  </div>
</div>
`;
